### ISSUE TITLE
Example code only produces one PNG

### ISSUE THREAD

**Issue by Wellington-Froelich** [RicDen/FlyByGen#1]
Hi Ric and everyone,

I was able to run FlyByGen, but I could not figure out how to display the animation. I went into each blender output (DustJet, Nucleus, Render, Scene, SpacecraftMotion) and clicked on View Animation under Render, but it just showed my a black screen. I didn't edit anything in the FlyByGen.py script so I am not sure where things went wrong (if they did at all!).

I'm guessing this issue is because I don't understand Blender well enough yet. Please let me know if it is. 

Thanks

Wellington

**Comment by RicDen** [comment]
Hey Wellington,

Thank you for reaching out.
Could you provide your latest .log file from the cache directory, just to make sure everything was executed correctly.
At which frames within the animation did you look? Generally, there are dark frames as shown in the presentation as the standard animation the object flys in and out of the field of view. It would also be good to know, if you see the object in the viewport shading -> solid mode similar to the picture below.

![image](https://github.com/RicDen/FlyByGen/assets/57988545/54f3f327-b56c-4a25-844a-8eb0b849eec2)

I am working on additional documentation, but the animation will only exist in the spacecraftMotion blender file or datasetGenerator.

**Comment by Wellington-Froelich** [comment]
[log_1.log](https://github.com/RicDen/FlyByGen/files/14450830/log_1.log)
Hi Ric

Yes I do see the object in the shading/solid display view. Here is what mine looks like:
![image](https://github.com/RicDen/FlyByGen/assets/36109279/4387a1fa-6310-42cf-8e5b-29502d9beab9)

I hit these buttons to bring up the animation:
![image](https://github.com/RicDen/FlyByGen/assets/36109279/7570f5cd-a59c-45dd-8760-cf9c2bc273a7)

Then it opens up a window and starts to cycle through frames: 
![image](https://github.com/RicDen/FlyByGen/assets/36109279/9987028d-19dd-461b-b6eb-cea201b9845b)

Sorry if I am jumping the gun on your documentation but if not, I hope this helps

**Comment by RicDen** [comment]
Hello, 

So I looked at your log. 
According to that you should fine some images in ``` 'C:\FLYBY_GEN\dataset\cache\SetUp_v1-1_Laptop_001\```.
There are a few important things to note:
1. The default set up starts rendering from frame 18, because that's when the object is for the first time in the field of view, so the object should be found among the pictures in the above given path.
2. My config documentation is not existing yet, but will come soon. The config files define among others all the paths. If you work in windows the relevant file for you is in  ```/src/config/windows/paths.json``` you can adapt the paths for this accordingly. Important are all the ones starting with a drive name like Z, D, C, etc. The rest is mostly related to automatic file naming or can be changed for different projects etc.
3. If you run the animation separately from one of the files, you might not start in frame 18 and thus you don't see the object until you render frame 18 or other images, which will include the object. 
4. With Numpad 0 you can see your current camera perspective, so when you press this, you should see if in any view if the object is in the field of view, no matter if solid, material or render preview alternatively select the correct view in the view setting.
![image](https://github.com/RicDen/FlyByGen/assets/57988545/bc3607db-7a7d-4c64-9722-ea551b983c76)

5. Other paths seem to work.
6. I have been working in blender 3.4 , 3.6.5 LTS and 3.6.9LTS, so I don't expect a particular problem with blender 3.6.0

So I am wondering, do you see the object in your standard output as state above? It took 17s to render one image, which is quite normal and indicates it rendered something and not just black.

**Comment by Wellington-Froelich** [comment]
Hey Ric,

When I hit Numpad 0 I get this image. It appears in the bottom left corner. 
![image](https://github.com/RicDen/FlyByGen/assets/36109279/d4fb8f81-25f8-46d0-81c6-ccadec2fd985)

I believe this translates to the frame 18 I see in the cache at `C:\FLYBY_GEN\dataset\cache\SetUp_v1-1_Laptop_001\`
![image](https://github.com/RicDen/FlyByGen/assets/36109279/8d9494c8-db37-4732-90ec-da09f5c90627)

So, just so I understand, I'll try to phrase what I think is happening. I have rendered frame 18 (it shows up in my cache). When I say 'View Animation' in the SpacecraftMotion Blender file it starts at Frame 0 and only goes to Frame 11 so I am not seeing anything in the animation. Is there another point that I missed?

Thanks

Wellington

**Comment by Wellington-Froelich** [comment]
Hi Ric

I took the time to re-read your last comment. I realize I don't fully understand your point #3. What does it mean to 'run the animation separately from the files'? First, are you referring to the files/pictures in the path you pointed out above? Second, How does one correctly run the animation?

Thanks

Wellington

**Comment by RicDen** [comment]
Hello, please excuse the delay. I have been busy with other projects and delivering the latest update. (Automatic parameter sweep and minor updates to the output path system)
You can always open any of the blender files and run the animation in Blender.
FlyByGen itself actually runs all the renders automatically and you will have all images needed for making a video in the output directory. 

I hope this welps

Cheers
Ric

**Comment by Wellington-Froelich** [comment]
Hi Ric

For me, the code only produces images (PNGs) for one frame, the one labeled 'start_frame' in the render.json file. No matter what I change in that file that I think might affect the output, the result is the same: only the 'start_frame' key is rendered in the final output.

Thanks

**Comment by RicDen** [comment]
Yes, that is the standard pipeline setup for the project's initialisation and testing of all submodules as quickly as possible.

For rendering multiple frames within an animation, please adapt in the src/config/blender/render.json the "start_frame" to the first frame you want to have rendered and the "end_frame" to the last frame you want to have rendered. The pipeline will then render all frames in between those values.
Below you see an example:
```
    "start_frame": 18,
    "end_frame" :20,
```

The first rendered image will be frame 18, second frame 19, and last image will be frame 20.
